legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -282,6 +310,71 @@ else if (dataSpace.supportInfo instanceof DataSpaceSupportCombinedInfo)
                         }
                         metamodel._supportInfo(supportInfo);
                     }
+                },
+                (dataSpace, context) ->
+                {
+                    // Move checking mapping/runtime utilized in the function within the curated template query does not align with the mapping/runtime applied in the execution context
+                    // to the sixth pass because function body and parameter are processed in the fifthPassBuilder
+                    // if we do the checking in the fifth pass, function body and parameter will be an empty list
```

Comment:
Seems we moved processing of functions from the second pass to the fifth pass in order to parallelize. 
Here: https://github.com/finos/legend-engine/pull/2918
Which is why we started failing the tests. 
Lets mention that PR in this comment.